CSharp/ArticleProjects/PersistOutputs/OutputFilesExample.cs

```diff
@@ -0,0 +1,131 @@
+﻿namespace Microsoft.Azure.Batch.Samples.Articles.PersistOutputs
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using WindowsAzure.Storage;
+    using WindowsAzure.Storage.Blob;
+    using Batch.Common;
+    using Common;
+
+    public class OutputFilesExample
+    {
+        public static CloudBlobContainer Run(
+            BatchClient batchClient,
+            CloudStorageAccount storageAccount,
+            string poolId,
+            int nodeCount,
+            string jobId)
+        {
+            const string containerName = "outputfilescontainer";
+
+            Func<ImageReference, bool> imageScanner = imageRef =>
+                imageRef.Publisher == "MicrosoftWindowsServer" &&
+                imageRef.Offer == "WindowsServer" &&
+                imageRef.Sku.Contains("2012-R2-Datacenter");
+
+            SampleHelpers.SkuAndImage skuAndImage = SampleHelpers.GetNodeAgentSkuReferenceAsync(batchClient, imageScanner).Result;
+
+            // Create and configure an unbound pool.
+            CloudPool pool = batchClient.PoolOperations.CreatePool(poolId: poolId,
+                virtualMachineSize: "standard_d1_v2",
+                targetDedicatedComputeNodes: nodeCount,
+                virtualMachineConfiguration: new VirtualMachineConfiguration(
+                    skuAndImage.Image,
+                    skuAndImage.Sku.Id));
+
+            // Commit the pool to the Batch service
+            GettingStartedCommon.CreatePoolIfNotExistAsync(batchClient, pool).Wait();
+
+            CloudJob job = batchClient.JobOperations.CreateJob(jobId, new PoolInformation { PoolId = poolId });
+
+            // Commit the job to the Batch service
+            job.Commit();
+            Console.WriteLine($"Created job {jobId}");
+
+            CloudBlobContainer container = storageAccount.CreateCloudBlobClient().GetContainerReference(containerName);
+            container.CreateIfNotExists();
+
+            string containerSas = container.GetSharedAccessSignature(new SharedAccessBlobPolicy()
+            {
+                Permissions = SharedAccessBlobPermissions.Write,
+                SharedAccessExpiryTime = DateTimeOffset.UtcNow.AddDays(1)
+            });
+            string containerUrl = container.Uri.AbsoluteUri + containerSas;
+
+            // Create a series of simple tasks which dump the task environment to a file and then write random values to a text file
+            IEnumerable<CloudTask> tasks = Enumerable.Range(1, 20).Select(i =>
+                {
+                    var taskId = i.ToString().PadLeft(3, '0');
+                    return new CloudTask(taskId, $"cmd /v:ON /c \"echo off && set && (FOR /L %i IN (1,1,100000) DO (ECHO !RANDOM!)) > output.txt\"")
```

Comment:
I really wish we could crack using the file conventions from the output files feature.  Customers shouldn't have to choose between the nice declarative approach of OutputFiles and the portal support of the conventions library.